examples/beta/openai/realtime/audio_stream_input_output_tools.py

```diff
@@ -0,0 +1,44 @@
+import asyncio
+from io import BytesIO
+
+from typing import AsyncGenerator, Any
+
+from pydub import AudioSegment
+from pydub.playback import play
+
+from mirascope.beta.openai import Realtime, record_as_stream, OpenAIRealtimeTool
+
+app = Realtime(
+    "gpt-4o-realtime-preview-2024-10-01",
+)
+
+
+def format_book(title: str, author: str) -> str:
+    return f"{title} by {author}"
+
+
+@app.receiver("audio")
+async def receive_audio(response: AudioSegment, context: dict[str, Any]) -> None:
+    play(response)
+
+
+@app.receiver("audio_transcript")
```

Comment:
we need to have a receiver that can receive the tool call

from openai docs: "The client must respond to the function call before by sending a conversation.item.create message with type: "function_call_output"."

this means we have to have a means of receiving the tool call and sending the tool call update in one fell swoop (i.e. it's both a receiver and a sender)

what if we did something like:

```python
@app.function_call(format_book)
async def handle_format_book(...) -> str: ...
```

the idea is that this function will handle when the API wants to call the `format_book` tool, and then we'll internally use the `str` response to then send the `function_call_output` event